dipper/sources/GWASCatalog.py

```diff
@@ -263,69 +318,73 @@ def _process_haplotype(
         mapped_genes = re.split(r';\s?', mapped_gene)
         snp_curies = list()
 
-        for index, snp in enumerate(snp_labels):
+        for snp in snp_labels:
             snp_curie, snp_type = self._get_curie_and_type_from_id(snp)
             if snp_type is None:
+                LOG.info('cant find type for SNP in %s', snp)
                 # make blank node
                 snp_curie = self.make_id(snp, "_")
+                model.addLabel(snp_curie, snp)
+            elif snp_curie[0] == '_':   # arrived an unlabeled blanknode
+                model.addLabel(snp_curie, snp)
 
             graph.addTriple(hap_id, self.globaltt['has_variant_part'], snp_curie)
             snp_curies.append(snp_curie)
 
         # courtesy http://stackoverflow.com/a/16720915
-        length = len(snp_labels)
+        # check lengths of mutiple lists
+        length = len(snp_curies)
         if not all(len(lst) == length
-                   for lst in [chrom_nums, chrom_positions, context_list]):
+                   for lst in [snp_labels, chrom_nums, chrom_positions, context_list]):
```

Comment:
This is evaluating as False when the condition is true, change to:

    if not all([len(lst) == length
               for lst in [snp_labels, chrom_nums, chrom_positions, context_list]]):